### PR TITLE
Wireguard package unit tests

### PR DESCRIPTION
**Pull request by alacuku**
# Description

The wireguard package exports a set of functions that allows to create and configure network interfaces of type wireguard.
This PR adds wrapper interfaces  for _wgctrl.Client interface_ and _netlink package_ in order to easily mock them. And it implements a mock version of these interfaces along the unit tests for the wireguard package.

# How Has This Been Tested?
Unit tests for the wireguard package.

### CODE REVIEW

File: pkg/liqonet/wireguard/wireguard_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			_, err := wireguard.NewWireguard(wgConfig, wgClient, n)
+			Expect(err).To(HaveOccurred())
+			return true
+		}, true),
+		Entry("Error while setting mtu to wireguard device", func() bool {
+			n := wireguard.NewNetLinkerFake(false, true, false)
+			_, err := wireguard.NewWireguard(wgConfig, wgClient, n)
+			Expect(err).To(HaveOccurred())
+			return true
+		}, true),
```
Comment (palexster) [inline pkg/liqonet/wireguard/wireguard_test.go:132]: Those two entries seem identical to me.

File: pkg/liqonet/wireguard/wireguard_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	})
+
+	DescribeTable("Creating new wireguard device", func(testFunc func() bool, expectedBool bool) {
+		outcome := testFunc()
+		Expect(outcome).Should(Equal(expectedBool))
+	},
+		Entry("Correctly creating a wireguard instance", func() bool {
+			n := wireguard.NewNetLinkerFake(false, false, false)
+			_, err := wireguard.NewWireguard(wgConfig, wgClient, n)
+			Expect(err).NotTo(HaveOccurred())
```
Comment (palexster) [inline pkg/liqonet/wireguard/wireguard_test.go:112]: Can we reinforce the assertions here by inspecting that the controller has correctly configured the n object?

Comment (alacuku) [reply pkg/liqonet/wireguard/wireguard_test.go:112]: the "n" object is a mocked interface. it is used instead the netlink package. we are not testing that the wireguard interface is created. just testing that the "wireguard.NewWireguard" method handles correctly the errors.

**Review by palexster — APPROVED** [review]

**Review by mlavacca — APPROVED** [review]

**Pull request merged**